deepin-movie reborn fail to compile with qtnetwork 5.13.0

https://gist.github.com/javashin/fe8a8bc7deb9c471081cc76d3d6eebcc

this is on gentoo any way to fix this ??

mkdir Build
                                                   cd Build
                                                   cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr ..
                                                   make
                                                   
-- The C compiler identification is GNU 9.1.0
-- The CXX compiler identification is GNU 9.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'dtkwidget'
--   Found dtkwidget, version 2.0.14
-- Checking for module 'dtkcore'
--   Found dtkcore, version 2.0.14
-- Checking for module 'mpv'
--   Found mpv, version 1.101.0
-- Checking for modules 'xcb;xcb-aux;xcb-proto;xcb-ewmh;xcb-shape'
--   Found xcb, version 1.13.1
--   Found xcb-aux, version 0.4.0
--   Found xcb-proto, version 1.13
--   Found xcb-ewmh, version 0.4.1
--   Found xcb-shape, version 1.13.1
-- Checking for modules 'libavformat;libavutil;libavcodec'
--   Found libavformat, version 58.20.100
--   Found libavutil, version 56.22.100
--   Found libavcodec, version 58.35.100
-- Checking for modules 'libffmpegthumbnailer;libpulse;libpulse-simple;openssl;dvdnav'
--   Found libffmpegthumbnailer, version 2.2.0
--   Found libpulse, version 12.2
--   Found libpulse-simple, version 12.2
--   Found openssl, version 1.1.0k
--   Found dvdnav, version 6.0.0
-- Checking for module 'libffmpegthumbnailer'
--   Found libffmpegthumbnailer, version 2.2.0
-- Configuring done
-- Generating done
-- Build files have been written to: /home/javashin/Desktop/deepin-movie-reborn-3.2.24/Build
Scanning dependencies of target json_i18n
[  0%] Built target json_i18n
Scanning dependencies of target deepin-movie_autogen
[  1%] Automatic MOC for target deepin-movie
[  1%] Built target deepin-movie_autogen
[  3%] Generating qrc_resources.cpp
Scanning dependencies of target deepin-movie
[  5%] Building CXX object src/CMakeFiles/deepin-movie.dir/deepin-movie_autogen/mocs_compilation.cpp.o
[  6%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/actions.cpp.o
[  8%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/dbus_adpator.cpp.o
[ 10%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/dmr_settings.cpp.o
/home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/dmr_settings.cpp: In function ‘QString dmr::flag2key(dmr::Settings::Flag)’:
/home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/dmr_settings.cpp:92:1: warning: control reaches end of non-void function [-Wreturn-type]
   92 | }
      | ^
[ 12%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/event_monitor.cpp.o
[ 13%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/event_relayer.cpp.o
[ 15%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/mainwindow.cpp.o
In file included from /usr/include/X11/Xlib.h:44,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/event_monitor.h:35,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:35:
/usr/include/qt5/QtNetwork/qocspresponse.h:64:5: error: expected identifier before numeric constant
   64 |     None = -1,
      |     ^~~~
/usr/include/qt5/QtNetwork/qocspresponse.h:64:5: error: expected ‘}’ before numeric constant
In file included from /usr/include/qt5/QtNetwork/QtNetwork:41,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/libdmr/online_sub.h:34,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/libdmr/player_engine.h:37,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:44:
/usr/include/qt5/QtNetwork/qocspresponse.h:63:1: note: to match this ‘{’
   63 | {
      | ^
In file included from /usr/include/X11/Xlib.h:44,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/event_monitor.h:35,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:35:
/usr/include/qt5/QtNetwork/qocspresponse.h:64:5: error: expected unqualified-id before numeric constant
   64 |     None = -1,
      |     ^~~~
In file included from /usr/include/qt5/QtNetwork/QtNetwork:41,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/libdmr/online_sub.h:34,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/libdmr/player_engine.h:37,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:44:
/usr/include/qt5/QtNetwork/qocspresponse.h:73:1: error: expected declaration before ‘}’ token
   73 | };
      | ^
/home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp: In constructor ‘dmr::MainWindow::MainWindow(QWidget*)’:
/home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:617:48: warning: ‘QSignalMapper::QSignalMapper(QObject*)’ is deprecated [-Wdeprecated-declarations]
  617 |     auto *signalMapper = new QSignalMapper(this);
      |                                                ^
In file included from /usr/include/qt5/QtCore/QtCore:196,
                 from /usr/include/qt5/QtWidgets/QtWidgetsDepends:3,
                 from /usr/include/qt5/QtWidgets/QtWidgets:3,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.h:37,
                 from /home/javashin/Desktop/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:32:
/usr/include/qt5/QtCore/qsignalmapper.h:56:28: note: declared here
   56 |     QT_DEPRECATED explicit QSignalMapper(QObject *parent = nullptr);
      |                            ^~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/deepin-movie.dir/build.make:318: src/CMakeFiles/deepin-movie.dir/common/mainwindow.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:100: src/CMakeFiles/deepin-movie.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

Confirmed this issue in openSUSE Tumbleweed.

```
[  157s] /usr/include/qt5/QtNetwork/qocspresponse.h:64:5: error: expected unqualified-id before numeric constant
[  157s]    64 |     None = -1,
[  157s]       |     ^~~~
[  157s] In file included from /usr/include/qt5/QtNetwork/QtNetwork:41,
[  157s]                  from /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/libdmr/online_sub.h:34,
[  157s]                  from /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/libdmr/player_engine.h:37,
[  157s]                  from /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:44:
[  157s] /usr/include/qt5/QtNetwork/qocspresponse.h:73:1: error: expected declaration before '}' token
[  157s]    73 | };
[  157s]       | ^
[  157s] /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp: In constructor 'dmr::MainWindow::MainWindow(QWidget*)':
[  157s] /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:617:48: warning: 'QSignalMapper::QSignalMapper(QObject*)' is deprecated [-Wdeprecated-declarations]
[  157s]   617 |     auto *signalMapper = new QSignalMapper(this);
[  157s]       |                                                ^
[  157s] In file included from /usr/include/qt5/QtCore/QtCore:196,
[  157s]                  from /usr/include/qt5/QtWidgets/QtWidgetsDepends:3,
[  157s]                  from /usr/include/qt5/QtWidgets/QtWidgets:3,
[  157s]                  from /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/mainwindow.h:37,
[  157s]                  from /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:32:
[  157s] /usr/include/qt5/QtCore/qsignalmapper.h:56:28: note: declared here
[  157s]    56 |     QT_DEPRECATED explicit QSignalMapper(QObject *parent = nullptr);
[  157s]       |                            ^~~~~~~~~~~~~
[  157s] /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/mainwindow.cpp:761:10: warning: unused variable 'mwpm' [-Wunused-variable]
[  157s]   761 |     auto mwpm = new MainWindowPropertyMonitor(this);
[  157s]       |          ^~~~
[  158s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/build'
[  158s] [ 82%] Built target dmr_test_autogen
[  158s] [ 83%] Building CXX object src/CMakeFiles/deepin-movie.dir/common/utility_x11.cpp.o
[  158s] cd /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/build/src && /usr/bin/c++  -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SQL_LIB -DQT_WIDGETS_LIB -DQT_X11EXTRAS_LIB -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/build/src -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/build/src/deepin-movie_autogen/include -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/build -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24 -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/ON -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/widgets -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/libdmr -I/home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/backends/mpv -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtCore -isystem /usr/lib/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtX11Extras -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtConcurrent -isystem /usr/include/qt5/QtDBus -isystem /usr/include/qt5/QtSql -isystem /usr/include/libdtk-2.0.6/DCore/../DWidget -isystem /usr/include/libdtk-2.0.6/DCore -isystem /usr/include/ffmpeg  -fomit-frame-pointer -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -std=c++11 -O3 -DNDEBUG   -fPIC -std=gnu++11 -o CMakeFiles/deepin-movie.dir/common/utility_x11.cpp.o -c /home/abuild/rpmbuild/BUILD/deepin-movie-reborn-3.2.24/src/common/utility_x11.cpp
[  159s] make[2]: *** [src/CMakeFiles/deepin-movie.dir/build.make:2757: src/CMakeFiles/deepin-movie.dir/common/mainwindow.cpp.o] Error 1
[  159s] make[2]: *** Waiting for unfinished jobs....
```

--- a/src/common/event_monitor.cpp
+++ b/src/common/event_monitor.cpp
@@ -32,8 +32,15 @@
 
 #include "event_monitor.h"
 #define Bool int
+#include <X11/Xlib.h>
 #include <X11/Xlibint.h>
 #include <X11/extensions/record.h>
+#undef Bool
+
+void callback(XPointer ptr, XRecordInterceptData* data)
+{
+    ((dmr::EventMonitor *) ptr)->handleRecordEvent(data);
+}
 
 namespace dmr {
 
@@ -85,13 +92,9 @@ void EventMonitor::run()
     }
 }
 
-void EventMonitor::callback(XPointer ptr, XRecordInterceptData* data)
+void EventMonitor::handleRecordEvent(void* dat)
 {
-    ((EventMonitor *) ptr)->handleRecordEvent(data);
-}
-
-void EventMonitor::handleRecordEvent(XRecordInterceptData* data)
-{
+    XRecordInterceptData* data = (XRecordInterceptData*)dat;
     if (!_recording) {
         XRecordFreeData(data);
         return;

--- a/src/common/event_monitor.h
+++ b/src/common/event_monitor.h
@@ -31,11 +31,6 @@
 #define EVENTMONITOR_H
 
 #include <QThread>
-#define Bool int
-#include <X11/Xlib.h>
-#include <X11/extensions/record.h>
-#undef Bool
-
 
 namespace dmr {
 
@@ -56,8 +51,7 @@ class EventMonitor : public QThread
 
 public:
     EventMonitor(QObject *parent = 0);
-    static void callback(XPointer trash, XRecordInterceptData* data);
-    void handleRecordEvent(XRecordInterceptData *);
+    void handleRecordEvent(void *);
 
     void resumeRecording();
     void suspendRecording();

--- a/src/libdmr/online_sub.h
+++ b/src/libdmr/online_sub.h
@@ -30,8 +30,10 @@
 #ifndef _DMR_ONLINE_SUB_H
 #define _DMR_ONLINE_SUB_H 
 
-#include <QtWidgets>
-#include <QtNetwork>
+#include <QFileInfo>
+#include <QNetworkAccessManager>
+#include <QNetworkRequest>
+#include <QNetworkReply>
 
 
 namespace dmr {

